text/0030-protocol-messages.md

```diff
@@ -0,0 +1,243 @@
++ Feature name: `protocol-messages`
++ Start date: 2020-04-27
++ RFC PR: [iotaledger/bee-rfcs#30](https://github.com/iotaledger/bee-rfcs/pull/30)
++ Bee issue: [iotaledger/bee#70](https://github.com/iotaledger/bee/issues/70)
+
+# Summary
+
+This RFC introduces the IOTA protocol messages.
+
+# Motivation
+
+To be able to take part in the IOTA networks, Bee nodes need to implement the exact same protocol presented in this RFC
+and currently being used by [IRI](https://github.com/iotaledger/iri) nodes and
+[HORNET](https://github.com/gohornet/hornet) nodes.
+
+# Detailed design
+
+## `Message` trait
+
+The `Message` trait is protocol agnostic and only provides serialization and deserialization to and from bytes buffers.
```

Comment:
typo "from byte buffers"

Reply:
It's on purpose, it's not a typo :)

Reply:
serialization -> to, deserialization -> from

Reply:
Ah you edited. Are you speaking about the 's' in bytes ?

Reply:
I am not a native English speaker (so someone who is might jump in here), but `bytes buffers` - even if not incorrect - sounds strange to me.

Reply:
Confirmed it's `byte buffer`, will edit, thanks!